blog/content/post/2024-02-07-cross2024_ftpm_tee_ta.md

```diff
@@ -0,0 +1,112 @@
+accessible from the Secure World. This mechanism supports running fTPM in the
+Secure World, enabling secure syscalls from user space. Secrets stored in fTPM
+are secure as long as the Secure Monitor is not compromised.
+
+![Cortex-A TrustZone Exception Levels](/img/TEE_ARM_Cortex-a_exception_levels.svg)
+![Cortex-A TrustZone](/img/TEE_ARM_Cortex-a.svg)
+
+Arm TrustZone also exists for the Cortex-m series but adopts a simpler and more
+hardware-focused approach relying on hardware mechanisms to manage the CPU
+state via interrupts.
+
+fTPM requires a non-trivial amount of computational
+resources and memory, which might be scarce in the environments where Cortex-M
+processors are typically used. Implementing fTPM could therefore be impractical
+due to the limited resources available on these devices. It's also rare for the
+Cortex-M devices demand the complex security functionalities that fTPM
+provides.
+
+![Cortex-M TrustZone](/img/TEE_ARM_Cortex-m.svg)
+
+## Fallbacks and Security Concerns
+
+_The best-protected systems have dedicated hardware security measures included
+from the beginning of their design process, starting with the specification
+for the processor core and the SoC infrastructure._
+
+As fTPM is implemented purely in software the huge advantage is that it can be
+added to already provisioned devices via an update. The caveat is that while
+this can improve the security of such devices there are hardware security
+concerns that the device should fulfill from the beginning. For example
+the [official OP-TEE Documentation specifies the Raspberry Pi 3 platform as not
+suitable for a secure implementation of Trusted Execution Environment](
+https://optee.readthedocs.io/en/latest/building/devices/rpi3.html#disclaimer).
+A sole CPU can't also provide features such as a good source of entropy, a
+secure counter, and a secure clock. These drawbacks can be mitigated but the
+manufacturer needs to seriously take into account potential vulnerabilities in
+the early stages of designing the embedded device.
+
+## Summary
+
+While traditional hardware TPMs offer robust physical security, fTPMs provide a
+a promising avenue for enhancing security in a variety of devices, from high-end
+computing systems to more constrained embedded devices.
+
+However, the transition to fTPMs is not without its challenges. This approach
+necessitates a careful evaluation of the underlying hardware's capacity to
+support secure operations.
+
+For more information about implementing the fTPM in practice please refer to the
+[fTPM: A Software-only Implementation of a TPM Chip](
+https://www.microsoft.com/en-us/research/publication/ftpm-software-implementation-tpm-chip/).
+This topic is also expanded at the [Securing Embedded Systems with
+fTPM implemented as Trusted Application in TEE](
+https://fosdem.org/2024/schedule/event/fosdem-2024-3097-securing-embedded-systems-with-ftpm-implemented-as-trusted-application-in-tee/)
```

Comment:
@tym2k1 is there a way to add to this blogpost a demo that you prepared for FOSDEM? I would add another section before summary with quick description and a video with that demo. If you do not see that possible at least please mention that it was presented at the linked presentation.